src/components/Swap/SwapButton/index.tsx

```diff
@@ -64,6 +66,16 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
   useEffect(() => setOpen((open) => (trade.trade ? open : false)), [trade.trade])
   // Close the review modal on chain change.
   useEffect(() => setOpen(false), [chainId])
+  // Close the review modal on defaults change
+  useEffect(
+    () => setOpen(false),
+    [
+      tokenDefaults?.defaultInputAmount,
+      tokenDefaults?.defaultInputTokenAddress,
+      tokenDefaults?.defaultOutputAmount,
+      tokenDefaults?.defaultOutputTokenAddress,
+    ]
+  )
```

Comment:
what do you mean by this? I'm seeing that the swap state (from `useSwapInfo`) includes things that change dynamically every few seconds - i.e. the best route, price impact, output amount - and we wouldn't want users to be flipped back to the inputs screen every time that happens

Reply:
think it would only make sense to flip users back to the inputs screen if the desired token input amount or token pair changes, which integrators can only change via the tokenDefaults